File: tests/fixtures/vulnerable/ws_no_auth.ts
// Vulnerable: Socket.IO without auth or origin check
import { Server } from "socket.io";

const io = new Server(httpServer);

io.on("connection", (socket) => {
  // No auth check on connect
  socket.on("join_room", (data) => {
    socket.join(data.room);
  });

  socket.on("send_message", (data) => {
    io.to(data.room).emit("message", data);
  });
});

File: tests/fixtures/vulnerable/ws_no_auth.py
"""Vulnerable: Socket.IO handlers without auth."""
from flask_socketio import SocketIO, emit

socketio = SocketIO()


@socketio.on("join_room")
def handle_join(data):
    """Bad: no auth check, no origin validation."""
    room = data["room"]
    join_room(room)
    emit("joined", {"room": room}, room=room)


@socketio.on("send_message")
def handle_message(data):
    """Bad: no per-message auth."""
    emit("message", data, room=data["room"])
